### ISSUE TITLE
Media Library input dont't show on Site logo block

### ISSUE THREAD

**Issue by AdsonCicilioti** [WordPress/gutenberg#39776]
### Description

Site logo block dont show the media library interface for upload a logo image.

my theme.json:
``` json
{
  "version": 2,
  "settings": {
    "appearanceTools": true,
    "color": {
      "defaultPalette": false,
      "defaultGradients": false,
      "palette": [
        {
          "slug": "primary",
          "color": "#0074c9",
          "name": "Primária"
        },
       ..... AND ANOTHER COLORS
```

### Step-by-step reproduction instructions

1. Go to Appearance > Editor 
2. Click on site logo block icon on header part

### Screenshots, screen recording, code snippet

![image](https://user-images.githubusercontent.com/6278398/160206849-16ef06cc-f53a-4a94-8a9c-96be7cf56f77.png)

### Environment info

- Wordpress 5.9.2, Gutenberg 12.8.1 
- Brave browser (Chromiun based)
- Linux - Arch Linux Distro

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by AdsonCicilioti** [comment]
![image](https://user-images.githubusercontent.com/6278398/160245344-e8cfe235-7a26-4ee4-8bb2-56e5c39ac54d.png)
![image](https://user-images.githubusercontent.com/6278398/160245380-62556a18-e6cb-4860-83a0-649085dd3105.png)

Probably something related to my nginx server.

It is worth remembering that my environment uses PHP-FPM 7.4, NGINX and Mysql.

So I'll investigate to see if my Nginx definitions are limiting some requests or something.

**Comment by AdsonCicilioti** [comment]
A directive I enabled in Nginx to make the `/wp-json/` route be rewritten to `/?rest_route=` somehow caused the `OPTIONS` requests made by the component to fail.

Fortunately I found that I don't need to override `/wp-json`, knowing that it won't work if permalinks are disabled, now I know that I always need to use `?rest_route=` which works in all cases.

This could be better documented, but ok.

Case solved. The problem was my environment.
